feature-libs/checkout/base/components/guards/checkout-steps-set.guard.ts

```diff
@@ -0,0 +1,133 @@
+import { Injectable, isDevMode } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  UrlTree,
+} from '@angular/router';
+import {
+  CheckoutDeliveryAddressFacade,
+  CheckoutDeliveryModesFacade,
+  CheckoutPaymentFacade,
+  CheckoutStep,
+  CheckoutStepType,
+} from '@spartacus/checkout/base/root';
+import { RoutingConfigService } from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { filter, map, switchMap, take } from 'rxjs/operators';
+import { CheckoutStepService } from '../services/checkout-step.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutStepsSetGuard implements CanActivate {
+  constructor(
+    protected checkoutStepService: CheckoutStepService,
+    protected routingConfigService: RoutingConfigService,
+    protected checkoutDeliveryAddressService: CheckoutDeliveryAddressFacade,
+    protected checkoutPaymentService: CheckoutPaymentFacade,
+    protected checkoutDeliveryModesService: CheckoutDeliveryModesFacade,
+    protected router: Router
+  ) {}
+
+  canActivate(route: ActivatedRouteSnapshot): Observable<boolean | UrlTree> {
+    let currentIndex = -1;
+    const currentRouteUrl = '/' + route.url.join('/');
+
+    // check whether the previous step is set
+    return this.checkoutStepService.steps$.pipe(
+      take(1),
+      switchMap((steps) => {
+        currentIndex = steps.findIndex((step) => {
+          const stepRouteUrl = `/${
+            this.routingConfigService.getRouteConfig(step.routeName).paths?.[0]
```

Comment:
```suggestion
            this.routingConfigService.getRouteConfig(step.routeName)?.paths?.[0]
```

when fixing the function for typing in `RoutingConfigService` for `getRouteConfig`

Reply:
Fixed